streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorContextImpl.java

```diff
@@ -71,7 +71,7 @@ public StateStore getStateStore(final String name) {
             return global;
         }
 
-        if (!currentNode().stateStores.contains(name)) {
+        if (currentNode.stateStores != null && !currentNode().stateStores.contains(name)) {
```

Comment:
I find this case a little confusing. `currentNode().stateStores == null` would seem to imply also that we don't have access to the store, or any store for that matter. Why do we allow this case to pass though?

Reply:
Yes, that's a bit messy. I'm going to change the way we get the processor context, and then we won't need this. Since there are already a few conflicts I'm also going to squash and rebase on current trunk.